### DISCUSSION TITLE
User authentication for admin users

### DISCUSSION THREAD

**Discussion by SoneNiko** [pocketbase/pocketbase#3890] [Q&A] [answered]
Hey, ive read some of the other threads here in search of an answer and I know this is totally not the supported use case but:

I would like to use the user authentication for the admins. Is there any way of doing that (no matter how janky or unintended). It would be really nice cus then i could use my organizations pre-existing SSO or even the gitlab accounts to authenticate my admins.

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the title.

Admins can authenticate only via email and password, there are no other options out of the box at the moment.

For the admin APIs, please check https://pocketbase.io/docs/api-admins/.

**Reply by SoneNiko** [reply]
Sorry, I could've formulated that more clearly. My problem is: it is a huge mgmt overhead to take care of the passwords of the admins. I would like to know if there is any way that I can use the auth providers (like gitlab or oidc or microsoft) for authentication of admins. I'm totally fine with janky solutions as long as I can automate the creation and deletion of admins. We use PB in a scenario where there are multiple people joining and leaving the team at the end of every quarter and automating this part would be really nice.

Thx in advance

**Reply by ganigeorgiev** [reply, marked as answer]
Unfortunately there is no easy way to do that at the moment.

It can be implemented, but I don't want to recommend it, nor I feel confident providing hacky solutions especially for the admins authentication.

Keep in mind that PocketBase admins are intended to be for super-users/developers type of accounts. Admins don't have roles and they can do everything (including deleting or creating other admins).

**Reply by SoneNiko** [reply]
Fair, I will open an issue as a proposal for an enhancement, that will allow people to fine tune the admins permissions and provide OAuth or LDAP as an authentication method. Thank you for your help.

Cheers

**Reply by ganigeorgiev** [reply]
_See https://github.com/pocketbase/pocketbase/issues/1388._

**Reply by SoneNiko** [reply]
Yeah i noticed later, thx
